writeups/Stonks.md

```diff
@@ -0,0 +1,104 @@
+# [Stonks](https://play.picoctf.org/practice/challenge/105?bookmarked=1&page=1)
+
+## Initial exploration
+After downloading the [C file](https://mercury.picoctf.net/static/a4ce675e8f85190152d66014c9eebd7e/vuln.c), I first tried running the netcat.
+
+```
+tiffanygan-picoctf@webshell:/tmp$ nc mercury.picoctf.net 59616
+Welcome back to the trading app!
+
+What would you like to do?
+1) Buy some stonks!
+2) View my portfolio
+1
+Using patented AI algorithms to buy stonks
+Stonks chosen
+What is your API token?
+test
+Buying stonks with token:
+test
+Portfolio as of Sat Aug 12 17:52:05 UTC 2023
+
+
+15 shares of CVP
+34 shares of X
+Goodbye!
+
+tiffanygan-picoctf@webshell:/tmp$ nc mercury.picoctf.net 59616
+Welcome back to the trading app!
+
+What would you like to do?
+1) Buy some stonks!
+2) View my portfolio
+2
+
+Portfolio as of Sat Aug 12 17:52:12 UTC 2023
+
+
+You don't own any stonks!
+Goodbye!
+```
```

Comment:
explain that we guessed that the important option will be (1) "Buy some stonks", also specifically the API token seems important, which helps us to find the corresponding part in the source code

Reply:
changed